### PR TITLE
Make the installer aware of the cluster

### PR DESCRIPTION
**Pull request by ark3**
Also improve error handling and add the remaining reports.

Definitely read this commit-by-commit...

### CODE REVIEW

File: cmd/edgectl/aes_install.go [hunk 1]
```diff
@@ -119,65 +119,107 @@ func aesInstall(cmd *cobra.Command, args []string) error {
...
+	}
+	aesManifests, err := getManifest("https://www.getambassador.io/yaml/aes.yaml")
+	if err != nil {
+		i.Report("fail_no_internet", ScoutMeta{"err", err.Error()})
+		return errors.Wrap(err, "download AES manifests")
+	}
+
+	// Figure out what version of AES is being installed
+	// TODO: Parse the manifests and build objects
+	// TODO: Extract version info from the Deployment object
```
Comment (alexgervais) [inline cmd/edgectl/aes_install.go]: Comment seems outdated

File: cmd/edgectl/aes_install.go [hunk 2]
```diff
@@ -119,65 +119,107 @@ func aesInstall(cmd *cobra.Command, args []string) error {
...
 	// Attempt to talk to the specified cluster
+	// TODO: Figure out cluster info to be passed along
 	i.kubeinfo = k8s.NewKubeInfo("", kcontext, "")
-
-	if err := i.ShowKubectl("cluster-info", "cluster-info"); err != nil {
+	if err := i.ShowKubectl("cluster-info", "", "cluster-info"); err != nil {
+		i.Report("fail_no_cluster")
 		return err
 	}
+	i.SetMetadatum("Cluster info", "cluster_info", "FIXME")
```
Comment (alexgervais) [inline cmd/edgectl/aes_install.go]: Probably needs to rebase this change with shared/2ksec

File: cmd/edgectl/aes_install.go [hunk 3]
```diff
@@ -168,7 +289,15 @@ func (i *Installer) Perform(kcontext string) error {
 	i.show.Printf("-> Installing the Ambassador Edge Stack %s\n", aesVersion)
 
 	// Attempt to talk to the specified cluster
-	// TODO: Figure out cluster info to be passed along
+	// TODO: Figure out cluster info to be passed along.
```
Comment (alexgervais) [inline cmd/edgectl/aes_install.go]: Most likely outdated.

File: cmd/edgectl/aes_install.go (hunk 1 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: The TODO comments are still valid, sadly.

File: cmd/edgectl/aes_install.go (hunk 2 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: Huh? GitHub marks this as outdated. It's not in my copy either.

File: cmd/edgectl/aes_install.go (hunk 3 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: Same here. GitHub has this marked as outdated.
image

**Pull request merged**